tests/unit/ui/components/TestConfirmButton.js

```diff
@@ -162,4 +166,11 @@ describe(__filename, () => {
     expect(root).not.toHaveClassName('ConfirmButton--show-confirmation');
     sinon.assert.calledWith(onConfirmSpy, event);
   });
+
+  describe('extractId', () => {
+    it('returns a unique ID provided by the ID prop', () => {
+      const id = 'special-button';
+      expect(extractId({ id })).toEqual(id);
```

Comment:
This needs to pass all component props like `extractId(getProps({ id: 'special-button' }))`.

It may not seem important now, but some refactoring later could require new props which might inadvertently break the test. Besides, you already have an abstraction to generate default props. It just needs a tweak like:

```diff
diff --git a/tests/unit/ui/components/TestConfirmButton.js b/tests/unit/ui/components/TestConfirmButton.js
index 81cdd4b62..48c84655a 100644
--- a/tests/unit/ui/components/TestConfirmButton.js
+++ b/tests/unit/ui/components/TestConfirmButton.js
@@ -14,16 +14,20 @@ import {
 import { dispatchClientMetadata } from 'tests/unit/amo/helpers';
 
 describe(__filename, () => {
-  const render = ({ children, i18n = fakeI18n(), ...props } = {}) => {
+  const getProps = ({ i18n = fakeI18n(), ...props } = {}) => {
+    return {
+      i18n,
+      id: 'Collection-confirm-delete',
+      message: 'some warning message',
+      onConfirm: sinon.stub(),
+      store: dispatchClientMetadata().store,
+      ...props,
+    };
+  };
+
+  const render = ({ children , ...otherProps } = {}) => {
     return shallowUntilTarget(
-      <ConfirmButton
-        i18n={i18n}
-        id="Collection-confirm-delete"
-        message="some warning message"
-        onConfirm={sinon.stub()}
-        store={dispatchClientMetadata().store}
-        {...props}
-      >
+      <ConfirmButton {...getProps(otherProps)}>
         {children || 'the default text of this button'}
       </ConfirmButton>,
       ConfirmButtonBase,
```